baseapp/streaming.go

```diff
@@ -0,0 +1,111 @@
+package baseapp
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+
+	"cosmossdk.io/store/streaming"
+	storetypes "cosmossdk.io/store/types"
+	"github.com/spf13/cast"
+
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+)
+
+const (
+	StreamingTomlKey                  = "streaming"
+	StreamingABCITomlKey              = "abci"
+	StreamingABCIPluginTomlKey        = "plugin"
+	StreamingABCIKeysTomlKey          = "keys"
+	StreamingABCIStopNodeOnErrTomlKey = "stop-node-on-err"
+)
+
+// RegisterStreamingServices registers streaming services with the BaseApp.
+func (app *BaseApp) RegisterStreamingServices(appOpts servertypes.AppOptions, keys map[string]*storetypes.KVStoreKey) {
+	// register streaming services
+	streamingCfg := cast.ToStringMap(appOpts.Get(StreamingTomlKey))
+	for service := range streamingCfg {
+		pluginKey := fmt.Sprintf("%s.%s.%s", StreamingTomlKey, service, StreamingABCIPluginTomlKey)
+		pluginName := strings.TrimSpace(cast.ToString(appOpts.Get(pluginKey)))
+		if len(pluginName) > 0 {
+			logLevel := cast.ToString(appOpts.Get(flags.FlagLogLevel))
+			plugin, err := streaming.NewStreamingPlugin(pluginName, logLevel)
+			if err != nil {
+				app.logger.Error("failed to load streaming plugin", "error", err)
+				os.Exit(1)
```

Comment:
Any reason for `os.Exit` over a panic with the error?

Reply:
Yes. We've had a conversation about this (ref: https://github.com/cosmos/cosmos-sdk/pull/13473#discussion_r1021917395). Note that If `panic` will ensure that the node will stop,  I would much rather send the error in a panic and clean this up.

Reply:
Either panic or os.exit seem misplaced.  Why not return the error and let the calling decide what to do?